tests/protocols/test_http.py

```diff
@@ -983,6 +997,20 @@ async def test_return_close_header(http_protocol_cls: HTTPProtocol):
     assert b"connection: close" in protocol.transport.buffer.lower()
 
 
+async def test_close_connection_with_multiple_requests(http_protocol_cls: HTTPProtocol):
+    app = Response("Hello, world", media_type="text/plain")
+
+    protocol = get_connected_protocol(app, http_protocol_cls)
+    protocol.data_received(REQEUST_AFTER_CONNECTION_CLOSE)
+    await protocol.loop.run_one()
+    assert b"HTTP/1.1 200 OK" in protocol.transport.buffer
+    assert b"content-type: text/plain" in protocol.transport.buffer
+    assert b"content-length: 12" in protocol.transport.buffer
+    # NOTE: We need to use `.lower()` because H11 implementation doesn't allow Uvicorn
+    # to lowercase them. See: https://github.com/python-hyper/h11/issues/156
+    assert b"connection: close" in protocol.transport.buffer.lower()
```

Comment:
This test passes without the changes on this PR. Can we have a better test?

Is it the first request that doesn't send the response, or the second? 🤔

Reply:
Ummm... well the test case is failing if I remove the change made with this PR.

Below are the results:
- with fix:
```sh
scripts/test tests/protocols/test_http.py -k test_close_connection_with_multiple_requests
tests/protocols/test_http.py ..                                                                                                        [100%]

===================================================== 2 passed, 118 deselected in 0.11s ======================================================
```
- without fix (commented the fix from `uvicorn/protocols/http/h11_impl.py` file):
```sh
scripts/test tests/protocols/test_http.py -k test_close_connection_with_multiple_requests
tests/protocols/test_http.py .F                                                                                                        [100%]

================================================ 1 failed, 1 passed, 118 deselected in 0.31s =================================================
```

The error raised by failed test case:
```
E           h11._util.LocalProtocolError: can't handle event type Response when role=SERVER and state=MUST_CLOSE
```

After the fix, the second request should not send the response, since the first request has the `Connection: close` header.